buildSrc/src/main/kotlin/org/jetbrains/DokkaBuildProperties.kt

```diff
@@ -0,0 +1,53 @@
+package org.jetbrains
+
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.jvm.toolchain.JavaLanguageVersion
+import org.jetbrains.kotlin.gradle.dsl.KotlinVersion
+import javax.inject.Inject
+
+/**
+ * Common build properties used to build Dokka subprojects.
+ *
+ * This is an extension created by the [org.jetbrains.conventions.Base_gradle] convention plugin.
+ *
+ * Default values are set in the root `gradle.properties`, and can be overridden via
+ * [CLI args, system properties, and environment variables](https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties)
+ */
+abstract class DokkaBuildProperties @Inject constructor(
+    private val providers: ProviderFactory,
+) {
+
+    /**
+     * The main version of Java that should be used to build Dokka source code.
+     *
+     * Updating the Java target is a breaking change.
+     */
+    val mainJavaVersion: Provider<JavaLanguageVersion> =
+        dokkaProperty("javaToolchain.mainCompiler") { JavaLanguageVersion.of(it) }
+
+    /**
+     * The version of Java that should be used to run Dokka tests.
+     *
+     * This value is set in CI/CD environments to make sure that Dokka still works with different
+     * versions of Java.
+     */
+    val testJavaLauncherVersion: Provider<JavaLanguageVersion> =
+        dokkaProperty("javaToolchain.testLauncher") { JavaLanguageVersion.of(it) }
```

Comment:
```suggestion
    val testJavaLauncherVersion: Provider<JavaLanguageVersion> =
        dokkaProperty("javaToolchain.testLauncher") { JavaLanguageVersion.of(it ?: "8") }
```

Reply:
I think it's best to have one place where the default value is defined, and that's in the root `gradle.properties` file. It could be annoying if someone updates it in one place, and doesn't realise it needs to also be updated in another.

Reply:
Make sense.